core/events/events_comment_base.ts

```diff
@@ -102,12 +100,10 @@ export class CommentBase extends AbstractEvent {
   ) {
     const workspace = event.getEventWorkspace_();
     if (create) {
-      const xmlElement = utilsXml.createElement('xml');
-      if (!event.xml) {
-        throw new Error('Ecountered a comment event without proper xml');
+      if (!event.json) {
+        throw new Error('Encountered a comment event without proper json');
       }
-      xmlElement.appendChild(event.xml);
-      Xml.domToWorkspace(xmlElement, workspace);
+      comments.append(event.json, workspace);
```

Comment:
BTW, this is why I preferred the older more verbose `Blockly.Foo.bar` syntax.  Took me a while to figure out what the heck 'comments' was.  Variable?  Nope.  Parameter?  Nope.  Global?  Nope.  ...

Reply:
I would take that up with JavaScript :P